### PR TITLE
Preflights-helm: run script

### PR DESCRIPTION
**Pull request by ggrosseveurgg**
Helm preflights script

Lot of options

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,3 +10,4 @@ Tools | Description
 ------------ | -------------
 [api-migration](./api-migration) | Facilitates the migration of incident remediation progress across different environments, including SaaS ↔ Self-Hosted, Self-Hosted ↔ Self-Hosted, and SaaS ↔ SaaS.
 [new-arch-migration](./new-arch-migration) | Assists in transitioning from the legacy GitGuardian architecture to the new architecture for Self-Hosted environments.
+[helm-preflights](./helm-preflights) | Used to ensure Gitguardian requirements are respected. It runs tests both from user and cluster space.
```
Comment (rjouhann) [inline README.md:13]: 💅 
```suggestion
[helm-preflights](./helm-preflights) | Ensures GitGuardian requirements are met prior installation or upgrade via [Helm on existing clusters](https://docs.gitguardian.com/self-hosting/installation/installation-existing-helm) by conducting tests from both the local user and the Kubernetes cluster.
```

**Comment by rjouhann** [comment]
@ggrosseveurgg could you add a README in the `helm-preflights` folder explaining how to use the tool?

**Comment by ggrosseveurgg** [comment]
> @ggrosseveurgg could you add a README in the `helm-preflights` folder explaining how to use the tool?

Done :)

File: helm-preflights/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Helm preflights
+
+Helm preflights are a set of tests that could be run anytime to ensure your cluster meets Gitguardian requirements.
```
Comment (rjouhann) [inline helm-preflights/README.md:3]:
```suggestion
The Helm preflights are a set of tests that can be run at any time to ensure your cluster meets Gitguardian's requirements.
```

File: helm-preflights/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Usage
+
+Among all the options, you can:
+- Define several values file
+- Define the namespace
+- Force retemplating (in case of values update)
+- Install `kubectl preflight` plugin
+
+You may run `./preflights --help` to see all options.
```
Comment (rjouhann) [inline helm-preflights/README.md:32]:
```suggestion
To view all options, run `./preflights --help`.
```

File: helm-preflights/preflights.sh [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    shift
+    VALUES_FILES+="-f $1 "
+    shift
+    ;;
+  -h | --help)
+    usage
+    exit 0
+    ;;
+  *) # positional argument
+    CHART=$1
```
Comment (ggguardian) [inline helm-preflights/preflights.sh:124]: You could add a test here to check that `CHART` var is not already set, and if this is the case raise an error:

At the moment, the script allows to specify multiple `args` and the last one will be used to define CHART....

**Comment by ggrosseveurgg** [comment]
I think this can be merged now, it's fine on templating side

**Comment by rjouhann** [comment]
> I think this can be merged now, it's fine on templating side

@ggrosseveurgg sounds good! we will merge it along with the `2024.4.0` release.

**Pull request merged**